contracts/SpokePool.sol

```diff
@@ -938,72 +949,69 @@ abstract contract SpokePool is
-                recipient: recipient,
-                relayer: exclusiveRelayer,
-                inputToken: inputToken.token,
-                outputToken: outputToken.token,
-                inputAmount: inputToken.amount,
-                outputAmount: outputToken.amount,
-                originChainId: originChainId,
-                destinationChainId: chainId(),
-                depositId: depositId,
-                fillDeadline: fillDeadline,
-                message: message
-            }),
+        // No need to override passed in destination chain id to be equal to chainId(),
+        // since this is not emitted in the FilledRelay event. So we do not have to fear a replay attack.
+
+        USSRelayExecutionParams memory relayExecution = USSRelayExecutionParams({
+            relay: relayData,
             relayHash: bytes32(0),
-            updatedOutputAmount: outputToken.amount,
-            updatedRecipient: recipient,
-            updatedMessage: message,
+            updatedOutputAmount: relayData.outputAmount,
+            updatedRecipient: relayData.recipient,
+            updatedMessage: relayData.message,
             repaymentChainId: repaymentChainId,
             slowFill: false,
             payoutAdjustmentPct: 0
         });
         relayExecution.relayHash = keccak256(abi.encode(relayExecution.relay));
 
-        // Validate RelayExecution data
-
-        // Pull output tokens from msg.sender and send to recipient
-
-        // Trigger `message` callback if appropriate.
+        _fillRelayUSS(relayExecution, msg.sender);
+    }
 
-        emit FilledUSSRelay(
-            inputToken,
-            outputToken,
-            repaymentChainId,
-            originChainId,
-            depositId,
-            fillDeadline,
-            exclusiveRelayer, // or msg.sender
-            depositor,
-            recipient,
-            message,
-            // updatedRecipient,
-            recipient,
-            // slowFill,
-            false,
-            // updatedOutputTokenAmount
-            outputToken.amount,
-            // payout adjustment pct
-            0,
-            // updatedMessage
-            message
+    /**
+     * @notice Request Across to send LP funds to this contract to fulfill a slow fill relay
+     * for a deposit in the next bundle.
+     * @dev Slow fills are not possible unless the input and output tokens are "equivalent", i.e.
+     * they route to the same L1 token via PoolRebalanceRoutes.
+     * @param relayData USSRelayData struct containing all the data needed to identify the deposit that should be
+     * slow filled. If any of the params are missing or different then Across will not include a slow
+     * fill for the intended deposit.
+     */
+    function requestUSSSlowFill(USSRelayData memory relayData) public override nonReentrant unpausedFills {
+        // TODO: Should we require that the exclusivity deadline has passed? In practice, I would never expect
```

Comment:
I think this raises a broader question: what happens if someone requests a slow fill _and_ the relay gets filled normally afterward? Is this effectively the same thing that happens today where a slow fill gets processed but it gets fast-filled before the slow fill can go through?

Reply:
I think that's the goal yeah. If a slow fill is requested but then followed by a fast fill in a subsequent bundle, then funds need to be returned to Hub. If the fast fill is in same bundle, nothing to do.

Reply:
do you understand it differently?

Reply:
Nope, that was my understanding, just making sure.

In my mind the slow fill is sort of the fallback path, so I wouldn't want exclusivity to be able to block a slow fill. In the worst case, if the exclusivity params are fat fingered, it would be nice to be able to fall back on slow fills for a stuck deposit. So I would argue for fewer limitations.

Thoughts?

Reply:
I'd agree that we don't really want restrictions on this function wrt exclusivity

Reply:
Removed this comment